src/nexmark/queries/q13.rs

```diff
@@ -0,0 +1,162 @@
+use super::NexmarkStream;
+use crate::{nexmark::model::Event, operator::FilterMap, Circuit, OrdZSet, Stream};
+
+use std::{
+    collections::HashMap,
+    fs::File,
+    io::{BufReader, Read, Result},
+};
+
+use csv;
+
+/// Query 13: Bounded Side Input Join (Not in original suite)
+///
+/// Joins a stream to a bounded side input, modeling basic stream enrichment.
+///
+/// TODO: use the new "filesystem" connector once FLINK-17397 is done
+/// ```
+/// CREATE TABLE side_input (
+///   key BIGINT,
+///   `value` VARCHAR
+/// ) WITH (
+///   'connector.type' = 'filesystem',
+///   'connector.path' = 'file://${FLINK_HOME}/data/side_input.txt',
+///   'format.type' = 'csv'
+/// );
+///
+/// CREATE TABLE discard_sink (
+///   auction  BIGINT,
+///   bidder  BIGINT,
+///   price  BIGINT,
+///   dateTime  TIMESTAMP(3),
+///   `value`  VARCHAR
+/// ) WITH (
+///   'connector' = 'blackhole'
+/// );
+///
+/// INSERT INTO discard_sink
+/// SELECT
+///     B.auction,
+///     B.bidder,
+///     B.price,
+///     B.dateTime,
+///     S.`value`
+/// FROM (SELECT *, PROCTIME() as p_time FROM bid) B
+/// JOIN side_input FOR SYSTEM_TIME AS OF B.p_time AS S
+/// ON mod(B.auction, 10000) = S.key;
+/// ```
+///
+/// NOTE: although the Flink test uses a static file as the side input, the
+/// query itself allows joining the temporal table from the filesystem file that
+/// is updated while the query runs, joining the temporal table using process
+/// time. Flink itself ensures that the file is monitored for changes. The
+/// [current documentation for this connector](https://nightlies.apache.org/flink/flink-docs-master/docs/connectors/table/filesystem/)
+/// is not that used above, since the new connector (in 1.17-SNAPSHOT) does not
+/// allow specifying a file name, but only a directory to monitor. Rather, the
+/// Nexmark test appears to have used the previous legacy filesystem connector, [stable filesystem connector](https://nightlies.apache.org/flink/flink-docs-release-1.15/docs/connectors/datastream/filesystem/)
+/// which does allow specifying a file path.
+///
+/// Also see [Flink's Join with a Temporal Table](https://nightlies.apache.org/flink/flink-docs-release-1.11/dev/table/streaming/joins.html#join-with-a-temporal-table).
+///
+/// So, although Flink supports monitoring the side-loaded file for updates, a
+/// simple static file is used for this bounded side-input for the Nexmark tests
+/// and that is also what is tested here.
```

Comment:
Not sure if this makes this query comparable... for example, we have no concept of joining using process time as per the above SQL query. We could implement this manually (updating the hash-map as new data is detected), but it's not needed if the benchmark doesn't use it.